### ISSUE TITLE
Improve visualization of side by side dragging

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#64214]
We support the ability to drag media blocks or media elements to the side of another media item to create quick side-by-side layouts.

Currently the effect is not super clear—see video:

https://github.com/user-attachments/assets/3711cba4-686d-4ee6-bbfe-c07cfdd6f8ae

It'd be nice if we could mimic the final result or approximate the effect by shrinking the image to make more space for the new one, or by expanding the drop zone pushing the other image slightly.

## Proposal

Here's a prototype by @seifeldinio::

https://github.com/user-attachments/assets/cba00f03-455b-4341-93d3-439253a47df2

Additionally, here's a slowed-down version to better illustrate the transition:

https://github.com/user-attachments/assets/90a3c517-347b-4f4e-aaff-971def21865f

**Comment by seifeldinio** [comment]
Something like this could be implemented. Here's a prototype for it:


https://github.com/user-attachments/assets/d3f352bb-dd46-421f-9df7-db57899d6ef8


When an image is dragged side by side, we can visualize the zone where the image will be placed. The adjacent image can shrink to make room for it, with a placeholder indicating that an image can be uploaded in that area.

Mockups:

![3](https://github.com/user-attachments/assets/6dadd29f-ef4e-4c13-a98e-e214ece4a149)

![4a](https://github.com/user-attachments/assets/aba7aa4a-86dd-436c-9c8b-2a6478106e81)

![1](https://github.com/user-attachments/assets/df35f284-2f7f-4c7a-94b9-847d690fc63e)


In the case of images stacked on top of each other, and when users select both images to arrange them in a row. 

When hovering over the "row" icon, a preview of the layout will be displayed. The default state returns when the hover is removed, and clicking the icon arranges the images side by side as previewed.

Here's a prototype for this:


https://github.com/user-attachments/assets/0c440b1e-ae6a-4979-ae96-be8bec3e3c37



Mockups:

![7](https://github.com/user-attachments/assets/74cc01cd-965a-4d54-b30e-3fbc1c28e9da)

![8](https://github.com/user-attachments/assets/e001cce6-70e3-46ed-a61a-9a7a587a70a5)

This approach enhances the visibility of system status and provides clear feedback for user interactions.

Would love to hear your thoughts!

**Comment by jasmussen** [comment]
Nice work, @seifeldinio! Those mockups seem to capture fairly well what was asked for in the original issue: a clearer preview of what will happen if you _drop_ what you're dragging. I wonder if this could be enhanced further with some animation, i.e. instead of the preview _instantly_ popping into the final layout, a transition might smooth out and aid the understanding of the preview. All animation is already disabled when prefers-reduced-motion is in place, that would also be the case here.

Many previews will likely cause layout shifts, where any content below the gallery moves upwards as the overall _height_ of the gallery gets shorter with the relatively smaller thumbnails. I wonder if there's any good way to eliminate or reduce this? The main risk of the layout shift is if the container block itself grows so small that the drop-zone itself changes size too, that might cause an infinite flickering loop. One idea: maintain the initial _height_ of the block, until you drop the image, and on mouse release animate the height into the place. The downside of this would be a slightly inaccurate preview.

Let's get some more thoughts on this, and thanks for the contribution. CC: @WordPress/gutenberg-design

**Comment by richtabor** [comment]
> I wonder if this could be enhanced further with some animation, i.e. instead of the preview instantly popping into the final layout, a transition might smooth out and aid the understanding of the preview.

+1

**Comment by seifeldinio** [comment]
How about something like this for the animation/smoother transition? It also shows how the layout shift can be mitigated: 


https://github.com/user-attachments/assets/e56a6870-6dd8-45e8-bb4b-5f59cd8dcc4b


Regarding the layout shift, I agree that maintaining a fixed height during the drag-and-drop action could work well. Even if the preview doesn't reflect the final layout in some cases, we could smooth out the transition from preview to actual layout with some animation to make it less jarring. 

Would love to hear your thoughts.

**Comment by jasmussen** [comment]
That looks great. Should there still be a vertical blue line indicator, at least in the initial moment before the actual animation kicks in?

**Comment by seifeldinio** [comment]
Yes for sure, the blue line indicator remains as it was. I didn't initially remove it, but it wasn't very visible before. I've made adjustments to make it more apparent now.

https://github.com/user-attachments/assets/0344f0be-2ff4-4ed1-9f35-67bf7bc9c655

**Comment by jasmussen** [comment]
Nice I actually meant showing the blue indicator right as you're hovering the right-most edge of the image, _before_ it starts scaling down. Sweet prototypes as always 🙏

**Comment by seifeldinio** [comment]
Thanks, @jasmussen! I've added the blue indicator to appear before the image starts scaling down. You can view the updated prototype here:


https://github.com/user-attachments/assets/cba00f03-455b-4341-93d3-439253a47df2


Additionally, here's a slowed-down version to better illustrate the transition:


https://github.com/user-attachments/assets/90a3c517-347b-4f4e-aaff-971def21865f


Please let me know if this meets your expectations!

**Comment by jasmussen** [comment]
Looks fantastic to me! The blue line can maybe disappear once the image actually expands, but that's not something we need to update in the prototype, next step surely is for a dev to take a look. Thanks for contributing!

**Comment by mtias** [comment]
This looks great, @seifeldinio, thanks! Agreed on dropping the blue line after the animation push.

**Comment by t-hamano** [comment]
This issue has been punted to 6.8 and has not seen any activity since. Let's move this issue to the 6.9 Editor Task Board and see what can be done in 6.9.

**Comment by ryanwelcher** [comment]
This was reviewed in the 6.9 bug scrub. @seifeldinio it seems that this had some great momentum before 6.8, is this something you would be able to finalize for 6.9? If not and no one else can pick it up, it may get punted to 7.0

**Comment by ryanwelcher** [comment]
Mentioned in the scrub today as a punt candidate

**Comment by t-hamano** [comment]
This issue was featured on today's Bug Scrub. Since no one is working on this issue and there has been no activity for a long time, let's punt it to 7.0.

**Comment by Mayank-Tripathi32** [comment]
I am working on implementations of the mockups since the designs have been finalised.

I will open the PR for the same.
